sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -278,14 +278,15 @@ private[state] class HDFSBackedStateStoreProvider(
 
   /** Initialize the store provider */
   private def initialize(): Unit = {
-    if (!fs.exists(baseDir)) {
-      fs.mkdirs(baseDir)
-    } else {
-      if (!fs.isDirectory(baseDir)) {
+    try {
+      if (!fs.getFileStatus(baseDir).isDirectory()) {
```

Comment:
Following the pattern, shouldn't the logic be inverted here?

- try to mkdir
- if it fails, re-throw with an informative message